protocol/storage/client.go

```diff
@@ -189,12 +189,7 @@ func (smc *Client) ProposeDeal(ctx context.Context, miner address.Address, data
 	// create payment information
 	totalCost := price.MulBigInt(big.NewInt(int64(pieceSize * duration)))
 	if totalCost.GreaterThan(types.ZeroAttoFIL) {
-		// The payment setup requires that the payment is mined into a block, currently we
-		// will wait for at most 5 blocks to be mined before giving up
-		ctxPaymentSetup, cancel := context.WithTimeout(ctx, 5*smc.api.BlockTime())
```

Comment:
I don't know where this came from, but it's not a good idea. This is actually timing out after 5 rounds, not 5 blocks. 5 consecutive null blocks won't be that uncommon. Also when testing with a short block time, this is a very short duration that can contribute to flaky tests.